### PR TITLE
[RFR] Products with API

### PR DESCRIPTION
**Pull request by djhi**
- [x] Product list fetches products from API
- [x] Better loading props
- [x] Product details fetches product from API

### Products
products

### Product Details
product

### CODE REVIEW

**Comment by djhi** [comment]
Damned, now I need fixtures in the test database for the tests to pass...

File: src/frontend/js/product/productSagas.spec.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+            expect(saga.next().value).to.deep.equal(call(fetchProducts));
+        });
+
+        it('should put the productsLoaded action with products on successfull fetch', () => {
+            const fetchProducts = sinon.spy();
+            const saga = loadProductsSaga(fetchProducts);
+
+            saga.next();
+            saga.next();
```
Comment (Kmaschta) [inline src/frontend/js/product/productSagas.spec.js]: There is two `saga.next()`, is it correct ?

**Comment by alexisjanvier** [comment]
@djhi could you use `e2e/lib/fixturesLoader.js` ?

File: src/frontend/js/product/productSagas.spec.js (hunk 1 above)
Comment (djhi) [inline src/frontend/js/product/productSagas.spec.js]: yes it is, we are using generators and here, we're testing the 3rd one

**Comment by djhi** [comment]
@alexisjanvier nope, we can't. No es6 in nightwatch tests, I tried

**Comment by alexisjanvier** [comment]
@djhi and with `babel-transformer` ?

**Comment by djhi** [comment]
I haven't tried with `babel-transformer`.

File: e2e/frontend/products.js [hunk 2]
```diff
@@ -1,27 +1,28 @@
 module.exports = {
     after: function(client) {
-        client.end();
+        // client.end();
```
Comment (Kmaschta) [inline e2e/frontend/products.js]: Commented code, but functional tests are turned off

**Comment by Kmaschta** [comment]
Do you have a screenshot for products ?

**Comment by djhi** [comment]
Updated with screenshots

**Comment by Kmaschta** [comment]
Nice !

**Pull request merged**